src/main/telemetry/ghst.c

```diff
@@ -112,10 +113,42 @@ void ghstFramePackTelemetry(sbuf_t *dst)
     sbufWriteU8(dst, 0x00);                     // tbd3
 }
 
+// GPS data, primary, positional data
+void ghstFrameGpsPrimaryTelemetry(sbuf_t *dst)
+{
+    // use sbufWrite since CRC does not include frame length
+    sbufWriteU8(dst, GHST_FRAME_GPS_PAYLOAD_SIZE + GHST_FRAME_LENGTH_CRC + GHST_FRAME_LENGTH_TYPE);
+    sbufWriteU8(dst, GHST_DL_GPS_PRIMARY);
+
+    sbufWriteU32(dst, gpsSol.llh.lat);
+    sbufWriteU32(dst, gpsSol.llh.lon);
+
+    // constrain alt. from -1 km to +10 km, send in units of meters
+    const uint16_t altitude = (constrain(getEstimatedAltitudeCm(), -1000 * 100, 10000 * 100) / 100) + 1000;
```

Comment:
Falling into the same trap as CRSF by including altitude here. It means that we can't supply altitude data (like from the barometer) unless the user also has a GPS.

Reply:
Reasonable point. I think the way to address this though would be to have a separate 'non-GPS' altitude value in a different packet. The OTA telemetry protocol has been set in stone for a while now, so difficult to break it out at this point. 
The current implementation does take care to send all three members of the '3d' position fix in a single telemetry packet, which avoids issues that occur when telemetry packets are dropped in marginal RF conditions.

Reply:
It's important to note that `getEstimatedAltitudeCm()` returns the altitude whether from GPS, baro, or a fusion of both (depending on user settings). So any other frame that might include altitude shouldn't be considered "non-GPS altitude". Also missing from this GPS data that's commonly useful to the user is "distance from home".

Reply:
Also this datatype is wrong since you're allowing negative numbers in your range constraint.